src/sentry/uptime/grouptype.py

```diff
@@ -1,15 +1,312 @@
 from __future__ import annotations
 
+import logging
 from dataclasses import dataclass
+from datetime import datetime
+from typing import override
 
+from django.utils import timezone as django_timezone
+from sentry_kafka_schemas.schema_types.uptime_results_v1 import CheckResult, CheckStatus
+
+from sentry import features, options
 from sentry.issues.grouptype import GroupCategory, GroupType
+from sentry.issues.issue_occurrence import IssueEvidence, IssueOccurrence
+from sentry.issues.status_change_message import StatusChangeMessage
 from sentry.ratelimits.sliding_windows import Quota
 from sentry.types.group import PriorityLevel
+from sentry.uptime.models import UptimeStatus, UptimeSubscription, get_project_subscription
 from sentry.uptime.types import (
     GROUP_TYPE_UPTIME_DOMAIN_CHECK_FAILURE,
     ProjectUptimeSubscriptionMode,
 )
-from sentry.workflow_engine.types import DetectorSettings
+from sentry.utils import metrics
+from sentry.workflow_engine.handlers.detector.base import DetectorOccurrence, EventData
+from sentry.workflow_engine.handlers.detector.stateful import (
+    DetectorThresholds,
+    StatefulDetectorHandler,
+)
+from sentry.workflow_engine.models import DataPacket, Detector
+from sentry.workflow_engine.processors.data_condition_group import ProcessedDataConditionGroup
+from sentry.workflow_engine.types import (
+    DetectorEvaluationResult,
+    DetectorGroupKey,
+    DetectorPriorityLevel,
+    DetectorSettings,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class UptimePacketValue:
+    """
+    Represents the value passed into the uptime detector
+    """
+
+    check_result: CheckResult
+    subscription: UptimeSubscription
+    metric_tags: dict[str, str]
+
+
+def build_detector_fingerprint_component(detector: Detector) -> str:
+    return f"uptime-detector:{detector.id}"
```

Comment:
I think we need to start sending this new fingerprint type along with issue occurrences now, so then we can backfill

Reply:
We have been actually, we use the same fingerprint here https://github.com/getsentry/sentry/blob/ad75d516fab4ffee7c7b06627b883738b8dd6bd2/src/sentry/uptime/issue_platform.py#L74